lib/PgSQL_Logger.cpp

```
@@ -43,24 +148,146 @@ static inline int write_encoded_length(unsigned char *p, uint64_t val, uint8_t l
+
+	if (other.username != nullptr) {
+		username = strdup(other.username);
+	}
+	if (other.schemaname != nullptr) {
+		schemaname = strdup(other.schemaname);
+	}
+	if (other.query_ptr != nullptr) {
+		size_t maxQueryLen = get_pgsql_eventslog_buffer_max_query_length_runtime();
+		size_t lenToCopy = std::min(other.query_len, maxQueryLen);
+		query_ptr = (char*)malloc(lenToCopy + 1);
+		memcpy(query_ptr, other.query_ptr, lenToCopy);
```

Cppcheck (2.19.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)